### PR TITLE
Ensure that ca override does not get lost

### PR DESCRIPTION
**Pull request by shinrich**
This is a fix on the feature added in PR #7130 

If the client request had a SNI name that both matched a server cert from ssl_multicert.config and the ca override feature in sni.yaml, the ca override data would be set first and then cleared by the openssl call SSL_set_SSL_CTX.

This PR augments the test to cover this case (select a non-default cert from ssl_multicert.config).  It also reapplies the ca override data after calling SSL_set_SSL_CTX.

Since this feature is in 9.0.x, the fix should go there as well.

### CODE REVIEW

File: iocore/net/SSLUtils.cc [hunk 1]
```diff
@@ -1146,19 +1148,25 @@ setClientCertLevel(SSL *ssl, uint8_t certLevel)
...
-  // X509_STORE_free() on the ca_certs store.  Hard to see how set0 could ever actually be useful, in
-  // what scenario would SSL objects never be freed?
-  //
-  SSL_set1_verify_cert_store(ssl, ca_certs);
-
+    // SSL_set_client_CA_list takes ownership of the STACK_OF(X509) structure
+    // So we load it each time to pass into the SSL object
+    if (file != nullptr && file[0] != '\0') {
+      SSL_set_client_CA_list(ssl, SSL_load_client_CA_file(file));
+    }
```
Comment (ywkaras) [inline iocore/net/SSLUtils.cc:1165]: I'm really lost with all this.  Doesn't this just set the _names_ of the CAs that are sent to the client, not the CA certs used to validate?  How is this relevant to Vinith's (or my) test, where the curl command only has a single cert it can send in any case?  Seems like there should be a way to avoid having to read a disk file on every single TLS handshake.

Comment (shinrich) [reply iocore/net/SSLUtils.cc:1165]: True.  I added this when debugging the issue.  Based on the documentation, it seemed like there were cases when the client would only look for appropriate intermediates if this was set.  Since we set this in the SSL_CTX case, I added it here.  Turns out it didn't fix the issue (at least not alone).  The ssl->cert->verify_store rewriting was the main problem.

File: iocore/net/SSLNetVConnection.cc [hunk 2]
```diff
@@ -924,6 +924,7 @@ void
 SSLNetVConnection::clear()
 {
   _serverName.reset();
+  _ca_cert_file.reset();
```
Comment (SolidWallOfCode) [inline iocore/net/SSLNetVConnection.cc:927]: Need a `_ca_cert_dir.reset();` also?

File: iocore/net/P_SSLNetVConnection.h [hunk 3]
```diff
@@ -519,6 +532,8 @@ class SSLNetVConnection : public UnixNetVConnection, public ALPNSupport, public
 
   // Null-terminated string, or nullptr if there is no SNI server name.
   std::unique_ptr<char[]> _serverName;
+  std::unique_ptr<char[]> _ca_cert_file;
+  std::unique_ptr<char[]> _ca_cert_dir;
```
Comment (SolidWallOfCode) [inline iocore/net/P_SSLNetVConnection.h:536]: These are needed because the SNI config can disappear during a transaction, destroying the original source, correct?

**Review by SolidWallOfCode — APPROVED** [review]

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.
